tests/test_bug_features.py

```diff
@@ -38,15 +39,25 @@
 
 @pytest.fixture
 def read(get_fixture_path):
-    def _read(path, feature_extractor_class, expected_results):
+    def _read(
+        path,
+        feature_extractor_class,
+        expected_results,
+        inline_data=None,
+    ):
         feature_extractor = feature_extractor_class()
 
-        path = get_fixture_path(os.path.join("bug_features", path))
+        if inline_data:
+            results = (feature_extractor(item) for item in inline_data)
+        else:
+            path = get_fixture_path(os.path.join("bug_features", path))
+            with open(path, "r") as f:
+                lines = f.readlines()
 
-        with open(path, "r") as f:
-            results = (feature_extractor(json.loads(line)) for line in f)
-            for result, expected_result in zip(results, expected_results):
-                assert result == expected_result
+            results = (feature_extractor(json.loads(line)) for line in lines)
```

Comment:
Why do we use `readlines()`? It adds the overhead of unnecessarily looping over all the lines in the file (as part of the `readlines()` method) and then looping over them again.